crates/evm/src/executor/inspector/utils.rs

```diff
@@ -25,12 +21,11 @@ macro_rules! try_or_continue {
 /// Get the address of a contract creation
 pub fn get_create_address(call: &CreateInputs, nonce: u64) -> Address {
     match call.scheme {
-        CreateScheme::Create => get_contract_address(b160_to_h160(call.caller), nonce),
+        CreateScheme::Create => call.caller.create(nonce),
         CreateScheme::Create2 { salt } => {
-            let salt = ru256_to_u256(salt);
-            let mut salt_bytes = [0u8; 32];
-            salt.to_big_endian(&mut salt_bytes);
-            get_create2_address(b160_to_h160(call.caller), salt_bytes, call.init_code.clone())
+            let init_code = alloy_primitives::Bytes(call.init_code.0.clone());
+            let init_code_hash = alloy_primitives::keccak256(init_code);
+            call.caller.create2(B256::from(salt), init_code_hash)
```

Comment:
Use [`create2_from_code`](https://docs.rs/alloy-primitives/latest/alloy_primitives/struct.Address.html#method.create2_from_code)

Reply:
done in https://github.com/foundry-rs/foundry/pull/5770/commits/274c2ea64a6eb8eab833f998b8252e09c864f945

Reply:
You don't need to clone the bytes

Reply:
I think you mean we should clone the init_code bytes? def don't need to access the internal `Bytes` but this is a move if we don't clone and won't compile (it should be a cheap clone anyway)

Reply:
Just borrow it brah

Reply:
ayyyy lmao 😅 , done in https://github.com/foundry-rs/foundry/pull/5770/commits/fb64d10a913a2951c7976d233449b36f9afe1efb